test/integration_helper.rb

```diff
@@ -122,7 +122,15 @@ def self.extend_object(obj)
     super
 
     base_port = 5000 + Process.pid % 100
-    Sinatra::Base.server.each_with_index do |server, index|
+    servers = Sinatra::Base.server.dup
+
+    # TruffleRuby doesn't support `Fiber.set_scheduler` yet
+    unless Fiber.respond_to?(:set_scheduler)
```

Comment:
Unfortunately, this check also skips falcon tests on MRI Ruby <3.0 

```
$ chruby-exec 2.7.7 -- ruby -e 'p Fiber.respond_to?(:set_scheduler)'
false
```

[falcon has an open requirement on async](https://rubygems.org/gems/falcon/versions/0.42.3), async since version 2.0 requires Ruby >=3.1.1 (https://rubygems.org/gems/async/versions/2.0.0). It should be possible to still run these tests on Ruby 2.x, don't you think @andrykonchin? It will just run falcon using async 1.x

Reply:
Right, so probably we should check explicitly for `RUBY_ENGINE == "truffleruby"` here.
TruffleRuby master has RUBY_VERSION 3.1 but doesn't support Fiber.set_scheduler yet.

Reply:
So:
```ruby
    if RUBY_ENGINE == "truffleruby" && !Fiber.respond_to?(:set_scheduler)
```

Reply:
I've prepared a PR with the fix (https://github.com/sinatra/sinatra/pull/1852).